File: Algorithms/Searching/Sequential/Python/sequential_search.py
"""

TODO: OOP implementation or something, idk.

"""
def elementSequentialSearch(l, search):
    for i in l:
        if i==search:
            found = True
        
    if (found):
        return ("{} found in list.".format(search))
    else:
        return ("{} not found in list.".format(search))

def indexSequentialSearch(l, search):
    for i in range(0,len(l)):
        if l[i]==search:
            found = True
        
    if (found):
        return ("{} found in list.".format(search))
    else:
        return ("{} not found in list.".format(search))

def main():
    list = []
    print("Enter integer elements in list.")
    print("Press q to stop entering numbers.")
    while(True):
        i = input()
        if(i == 'q'):
            break
        list.append(i)

    n = input("Enter a number to search in list: ")
    print(elementSequentialSearch(list,n))

main()

File: Algorithms/Searching/Sequential/Javascript/sequential_search.js
function main() {
    let list = [];
    let found = false;

    console.log("Enter integer elements in list: ");
    while (1) {
        let elements = prompt();
        if (elements === "q") {
            break;
        } else {
            list.push(elements);
        }
    }

    let search = prompt("Enter the number to search: ");

    list.forEach((l) => {
        if (l === search) {
            found = true;
        }
    });

    if (found === false) {
        return "Not found";
    } else {
        return "Found";
    }
}

main();
